oreo.go

```diff
@@ -218,6 +220,28 @@ func (c *Client) saveCookies(resp *http.Response) error {
 		cookies = mergedCookies
 	}
 
+	lockFile := fmt.Sprintf("%s.lock", c.cookieFile)
```

Comment:
`lock.TryLock` will generate file if it is missing, then we delete the lock file via defer when done with it.  It will be right next to the configured cookie file, just with a `.lock` file suffix.